src/igvfd/types/file_set.py

```diff
@@ -1196,6 +1234,32 @@ class ConstructLibrarySet(FileSet):
     def applied_to_samples(self, request, applied_to_samples):
         return paths_filtered_by_status(request, applied_to_samples)
 
+    @calculated_property(
+        condition='applied_to_samples',
+        schema={
+            'title': 'File Sets',
+            'description': 'The file sets with samples that integrate this construct library set.',
+            'type': 'array',
+            'minItems': 1,
+            'uniqueItems': True,
+            'items': {
+                'title': 'File Set',
+                'type': ['string', 'object'],
+                'linkTo': 'FileSet',
+            },
+            'notSubmittable': True
+        })
+    def file_sets(self, request, applied_to_samples=None):
+        file_sets = []
+        for sample in applied_to_samples:
+            sample_object = request.embed(sample,
+                                          '@@object_with_select_calculated_properties?'
+                                          'field=file_sets'
+                                          )
+            file_sets = file_sets + sample_object.get('file_sets', [])
+        if file_sets:
+            return file_sets
```

Comment:
do you think set() would need to be applied to file_sets at the end? (I suppose multiple samples could be linked to same analysis set?)

Reply:
Actually I will remove this calculated property, I realize it is unnecessary since I have embedded file sets from the samples.